homeassistant/components/sensor/google_travel_time.py

```diff
@@ -52,10 +52,10 @@
     vol.Required(CONF_DESTINATION): cv.string,
     vol.Required(CONF_ORIGIN): cv.string,
     vol.Optional(CONF_NAME): cv.string,
-    vol.Optional(CONF_TRAVEL_MODE): vol.In(TRAVEL_MODE),
+    vol.Optional(CONF_TRAVEL_MODE): cv.string,
```

Comment:
We should still not overload mode. Have a travel_mode and a travel_mode_entity.

Reply:
You can then also put the correct validations on it. Like making the 2 options exclusive

Reply:
Ok. So it's fine to define a vol.Required as cv.string, because it's got to be there but it can be either a sensor or a string? But an Optional should be one of two parameters, rather than have exactly the same logic to determine if it is an entity? The class attributes are not overloaded, it's just the params that would go in configuration.yaml.

Reply:
They won't both be string, as the one pointing at the entity would validate as `cv.entity_id`.